app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,20 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def facebook
+    # You need to implement the method below in your model (e.g. app/models/user.rb)
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      self.current_user = @user
```

Comment:
You don't need to assign the `current_user` here. Devise should take care of this for you.